internal/manager/conditions_crd.go

```diff
@@ -0,0 +1,43 @@
+package manager
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/controllers/utils"
+)
+
+// CRDControllerCondition determines if a resource controller for a given CRD should be enabled.
+type CRDControllerCondition struct {
+	gvr           schema.GroupVersionResource
+	toggleEnabled bool
+	restMapper    meta.RESTMapper
+	log           logr.Logger
+}
+
+func NewCRDCondition(gvr schema.GroupVersionResource, enabled bool, restMapper meta.RESTMapper) CRDControllerCondition {
+	return CRDControllerCondition{
+		toggleEnabled: enabled,
+		gvr:           gvr,
+		restMapper:    restMapper,
+		log: ctrl.Log.WithName("crd_controller_condition").
+			WithValues("group", gvr.Group, "version", gvr.Version, "resource", gvr.Resource),
+	}
+}
+
+func (c CRDControllerCondition) Enabled() bool {
+	if c.toggleEnabled {
+		if !utils.CRDExists(c.restMapper, c.gvr) {
+			c.log.Info(fmt.Sprintf("Disabling the '%s' controller due to missing CRD installation", c.gvr.Resource))
```

Comment:
You can use `infof` instead of `info(fmt.Sprintf())` to use format directives. Furthermore, log lines should never be capitalized.

Reply:
`ctrl.Log` doesn't provide `Infof` unfortunately. [a1b04b7](https://github.com/Kong/kubernetes-ingress-controller/pull/3013/commits/a1b04b7b3525d2c833c1766103cd0625708f0a1a)